linera-core/src/unit_tests/test_utils.rs

```diff
@@ -301,22 +310,31 @@ where
     ) -> Result<(), Result<ChainInfoResponse, NodeError>> {
         let mut validator = self.client.lock().await;
         let mut notifications = Vec::new();
+        let handle_certificate_result = match validator.fault_type {
```

Comment:
We could probably deduplicate this with a `LocalValidator::handle_certificate` method.

Reply:
I'm not sure I get this one 🤔